Lib/test/test_clinic.py

```diff
@@ -694,17 +689,19 @@ def test_c_name(self):
         self.assertEqual("os_stat_fn", function.c_basename)
 
     def test_cloning_nonexistent_function_correctly_fails(self):
-        stdout = self.parse_function_should_fail("""
-            cloned = fooooooooooooooooooooooo
+        block = """
+            cloned = fooooooooooooooooo
             This is trying to clone a nonexistent function!!
+        """
+        err = "Couldn't find existing function 'fooooooooooooooooo'!"
+        with support.captured_stdout() as out:
+            self.expect_failure(block, err, lineno=0)
+        expected_debug_print = dedent("""\
+            cls=None, module=<FakeClinic object>, existing='fooooooooooooooooo'
+            (cls or module).functions=[]
         """)
-        expected_error = """\
-cls=None, module=<FakeClinic object>, existing='fooooooooooooooooooooooo'
-(cls or module).functions=[]
-Error on line 0:
-Couldn't find existing function 'fooooooooooooooooooooooo'!
-"""
-        self.assertEqual(expected_error, stdout)
+        out = out.getvalue()
+        self.assertIn(expected_debug_print, out)
```

Comment:
Maybe we should just do this to `clinic.py`, so that the debug prints go to the same place as the error message? The debug prints are supposed to help you figure out why you've hit the error, after all :p

```diff
--- a/Tools/clinic/clinic.py
+++ b/Tools/clinic/clinic.py
@@ -4720,8 +4720,8 @@ def state_modulename_name(self, line: str) -> None:
                     if existing_function.name == function_name:
                         break
                 else:
-                    print(f"{cls=}, {module=}, {existing=}")
-                    print(f"{(cls or module).functions=}")
+                    print(f"{cls=}, {module=}, {existing=}", file=sys.stderr)
+                    print(f"{(cls or module).functions=}", file=sys.stderr)
                     fail(f"Couldn't find existing function {existing!r}!")
```

Reply:
+1 stderr ftw